src/frontend/src/eth/services/eth-transactions-batch.services.ts

```diff
@@ -0,0 +1,51 @@
+import { ETHERSCAN_MAX_CALLS_PER_SECOND } from '$env/rest/etherscan.env';
+import { loadEthereumTransactions } from '$eth/services/eth-transactions.services';
+import { batch } from '$lib/services/batch.services';
+import type { Token, TokenId } from '$lib/types/token';
+import type { ResultSuccess } from '$lib/types/utils';
+
+export type ResultByToken = ResultSuccess & { tokenId: TokenId };
+
+type PromiseResult = Promise<ResultByToken>;
+
+export const batchLoadTransactions = ({
+	tokens,
+	tokensAlreadyLoaded
+}: {
+	tokens: Token[];
+	tokensAlreadyLoaded: TokenId[];
+}): AsyncGenerator<PromiseSettledResult<ResultByToken>[]> => {
+	const promises = tokens.reduce<PromiseResult[]>(
+		(acc, { network: { id: networkId }, id: tokenId }) => {
+			if (tokensAlreadyLoaded.includes(tokenId)) {
+				return acc;
+			}
+
+			const promise = (async (): PromiseResult => {
```

Comment:
yes, it can be done both ways actually, but if i don't execute it here (i don't create the promise but only the function), then I think I need to change the batch util to be
`const results = await Promise.allSettled(batch.map((fn) => fn())`

Reply:
I wonder if the later would not be nicer? But not a show stopper for this PR.